judge/models/contest.py

```diff
@@ -338,16 +344,19 @@ def access_check(self, user):
 
             in_org = self.organizations.filter(id__in=user.profile.organizations.all()).exists()
             in_users = self.private_contestants.filter(id=user.profile.id).exists()
+        else:
+            in_org = False
+            in_users = False
 
-            if not self.is_private and (self.is_organization_private or self.is_private_viewable):
-                if in_org:
-                    return
-                raise self.PrivateContest()
+        if not self.is_private and (self.is_organization_private or self.is_private_viewable):
+            if in_org:
+                return
+            raise self.PrivateContest()
 
-            if self.is_private and not self.is_organization_private:
-                if in_users:
-                    return
-                raise self.PrivateContest()
+        if self.is_private and not (self.is_organization_private or self.is_private_viewable):
+            if in_users:
+                return
+            raise self.PrivateContest()
 
             if self.is_private and (self.is_organization_private or self.is_private_viewable):
```

Comment:
Indentation is still the wrong level. Also you're comparing `self.is_private and (self.is_organization_private or self.is_private_viewable)` twice? (Once here and once on line 356).

Reply:
This line and the one on 356 are not the same.